feature/job-event

dalawton opened the pull request:
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Added a new `!jobs` command to the Discord bot, allowing users to search for jobs and internships with optional filters.
  * Job search results are formatted for easy reading and limited to 10 entries, with a summary if more are available.
  * Help command updated to describe the new `!jobs` feature.

* **Bug Fixes**
  * Improved CSV file reading to ensure consistent UTF-8 encoding.

* **Tests**
  * Introduced comprehensive tests for job search, filtering, and formatting functionalities.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Lementknight commented:
@coderabbitai review

dalawton commented:
@coderabbitai review

Lementknight commented:
@coderabbitai full review

dalawton commented:
@coderabbitai full review

alantomw reviewed (approved):
Great job so far Dani! Your solution works great for now, but you will mostly likely have to change functionality for Aryan's NLP thing later on. 

Some minor notes:
adding a break statement after 
                if each_term.lower() in each_field.lower():
                    include_job = True
                    ## break would go here
in job_event.py would make search more efficient
(also why is it named job_event? a little confusing)

for your job command functionality, make sure to catch RuntimeError along with OSError

Good job again, this is perfect for an MVP

alantomw commented:
ya didnt know how to do that 👍

Lementknight reviewed (approved):
LGTM

andewmark reviewed (changes requested):
Please review my suggested changes, and I have one question.

Do the flags change anything about what is being searched on? It seems as though the logic is just ignoring the flags and looking only for the filters. If that's how you wanted this to be implemented, that's completely fine, but I am questioning as to why we are checking for flags if the user isn't expected to filter on them in the first place.

dalawton commented:
> @dalawton Please review my suggested changes, and I have one question.
> 
> Do the flags change anything about what is being searched on? It seems as though the logic is just ignoring the flags and looking only for the filters. If that's how you wanted this to be implemented, that's completely fine, but I am questioning as to why we are checking for flags if the user isn't expected to filter on them in the first place.

@andewmark  originally I had the flags and created tests for those however I changed my mind recently and removed the flag filtering option. I seem to have forgotten to update the test files when I changed the logic. I will update the test files asap.

dalawton commented:
@andewmark I removed the test cases that are no longer applicable, in doing so I believe the remaining test cases are sufficient for coverage of the files. Please let me know if you disagree.

andewmark commented on data_processing/job_event.py:
```diff
@@ -0,0 +1,149 @@
+)
+
+
+def paste_jobs_command(command_args: str) -> str:
+    """
+    Parses the jobs command arguments to extract search filters
+        from flag notation
+
+    Args:
+        command_args (str): The arguments passed to the command,
+                                as a string.
+
+    Returns:
+        str: string containing parsed parameters
+    """
+    params = ""
+    if not command_args:
+        return params
+    args = command_args.split()
+    for term in args:
+        if term.startswith("-"):
+            pass
+        else:
+            params += term + " "
+    return params.strip()
```
@dalawton Gotcha. 

Your jobs works as intended, but now this method starts to seem arbitrary. It's looking for a flag indicator, but we never give any instruction to the user to input a flag.
```suggestion
def paste_jobs_command(command_args: str) -> str:
    """
    Returns the command line arguments exactly as passed in,

    Args:
        command_args (str): The arguments passed to the command.

    Returns:
        str: The arguments string as-is, stripped of leading/trailing whitespace.
    """
    if not command_args:
        return ""
    return command_args.strip()
```

You can also remove this method and move the logic into `bot.py`
